### DISCUSSION TITLE
How to use last demo on Async

### DISCUSSION THREAD

**Discussion by zhouzhang829** [acheong08/EdgeGPT#134] [General]
How can I use the last dev demo Async for better visual in my terminal?
Do I have to write a new script or something?

**Comment by hrizal** [comment]
as an alternative way of accessing with one question

1. install edgegpt and create cookies file from chrome
as described in README.md
check : 

> - Checking access
> - Getting authentication
> 

2.  copy paste this script, save as edgegpt-api.py
```
import asyncio
import sys
import json
from EdgeGPT import Chatbot, ConversationStyle

if len(sys.argv) > 1:
    prompt = sys.argv[1]
else:
    sys.exit

async def main():
    bot = Chatbot('./cookies.json')
    result = await bot.ask(prompt=prompt, conversation_style=ConversationStyle.creative)   
    print(result['item']['messages'][1]['text'])
    await bot.close()

if __name__ == "__main__":
    asyncio.run(main())
```

use from command line (CLI)
`python3 edgegpt-api.py "your question here"`

You can wrap it in another programming language

**Comment by tianlichunhong** [comment]
Yes, this should work.But every time run "python3 edgegpt-api.py", need to import and load cookies.json， this need several second. It will be not very good experiense. Right?

**Reply by hrizal** [reply]
the time to import and load cookies is insignificant,
but indeed getting answer from chatgpt takes seconds

I'm using it as a chatbot, and users willing to wait for a satisfactory answer 😅

i am also combining with edgegpt, if chatgpt cant answer

**Reply by acheong08** [reply]
Loading cookies take seconds? It's just reading a file and parsing it.

I think the issue is that the initial handshake takes a while due to the websocket module.
